File: content/Integer Error/CS2 Java/securityChecklist.java
import java.util.Scanner;

public class PopCheck {
  public static void main(String[] args) {
    Scanner scan = new Scanner(System.in);
    System.out.print("What is the current population? ");
    short pop = scan.nextShort();
    System.out.print("What is the rate of growth?
                     (e.g., for 10% enter 10) ");
    short growth = scan.nextShort();
    float growthRate = growth / 100f;
    System.out.println("Year\tGrowth\tNew Population");
    for (int i = 1; i <= 10; i++) {
      short increase = (short) (pop * growthRate);
      short newpop = (short) (pop + increase);
      System.out.println(i+"\t"+increase+"\t" + newpop);
      pop = newpop;
    }
    System.err.println("Final population is " + pop);
  }
}

File: content/Integer Error/CS2 C++/securityChecklist.cpp
#include <iostream>
using namespace std;

int main() {

  short pop;
  short growth;

  cout << "What is the current population? ";
  cin >> pop;
  cout << "What is the rate of growth? (e.g., for 10% enter 10)";
  cin >> growth;

  float growthRate = static_cast(growth) / 100;
  cout << growthRate << endl;

  cout << "Year\tGrowth\tNew Population" << endl;

  for (size_t i = 1; i <= 10; i++) {
    short increase = static_cast(pop * growthRate);
    short newpop = pop + increase;

    cout << i << "\t" << increase << "\t" << newpop << endl;
    pop = newpop;
  }
  cout << "Final population is " << pop << endl;
}